requests/utils.py

```diff
@@ -645,11 +645,12 @@ def is_ipv4_address(string_ip):
         return False
     for octet in ip_split:
         try:
-            int(octet)
+            tmp = int(octet)
```

Comment:
> We already have a check for valid ip that works (is_valid_ipv4)
Sure, but it's better to use standard methods that work properly instead of reinventing the wheel.

The current version of  `is_ipv4_address` still doesn't handle all the edge cases - for example, it returns `True` for an ip of `'08.08.08.08'`.

The `ipaddress` module has functions to check given IPs and networks. This could probably be used for both IP and CIDR checking.

Reply:
The ipaddress module is also fallible, `ipaddress.IPv4Address('01.01.01.01')` returns successfully.

Reply:
The ipaddress module doesn't exist for Python 2.7